lib/train/transports/vmware.rb

```diff
@@ -0,0 +1,190 @@
+# encoding: utf-8
+require 'train/plugins'
+require 'open3'
+require 'ostruct'
+require 'json'
+require 'mkmf'
+
+module Train::Transports
+  class VMware < Train.plugin(1)
+    name 'vmware'
+    option :viserver, default: ENV['VISERVER']
+    option :username, default: ENV['VISERVER_USERNAME']
+    option :password, default: ENV['VISERVER_PASSWORD']
+    option :insecure, default: false
+
+    def connection(_ = nil)
+      @connection ||= Connection.new(@options)
+    end
+
+    class Connection < BaseConnection # rubocop:disable ClassLength
+      POWERSHELL_PROMPT_REGEX = /PS\s.*> $/
+
+      def initialize(options)
+        super(options)
+
+        options[:viserver] = options[:viserver] || options[:host]
```

Comment:
Fun shorthand: try `||=`.  Like `+=`, but for or.  It's great for setting defaults.
```
options[:viserver] ||= options[:host]
     
```

Reply:
Indeed it is, will do!